mitodl/open-discussions
# Enable channel-specific google analytics tracking

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #997 

#### What's this PR do?
- adds a GA tracking id to the `Channel` model (`ga_tracking_id`)
- adds a simple admin page to allow manual entry of GA tracking id's for channels
- modifies the Django `index` view to include a dict of channel:tracking_id keys/values in `js_settings`
- modifies the `withTracker` React component to:
  -  initialize a GA tracker for each channel that has a tracker id
  - record a pageview with the appropriate tracker id if the current page path includes a tracked channel name.

#### How should this be manually tested?
- Turn off your ad blockers
- Set `GA_TRACKING_ID` to some value in your .env file
- Go to `admin/channels/channel/` and enter a value (real or dummy) for `ga_tracking_id` for 1+ channels
- With the network developer tab open, go to the home page.  You should see one request made to `https://www.google-analytics.com/collect` with request parameter `tid=<GA_TRACKING_ID>`
- Go to a channel page (or post page) for a channel with no `ga_tracking_id`.  Again there should be only one request made to `https://www.google-analytics.com/collect`, with request parameter `tid=<GA_TRACKING_ID>`
- Go to a channel page, post detail page, and comments page for a channel that you did enter a `ga_tracking_id` value for.  In addition to the usual GA pageview request as above, there should be one with request parameter `tid=<Channel.ga_tracking_id>` on each of these pages.
- If you happened to use real tracking id's, go to that GA account's "real-time" view and check that the data was received.
